dataactbroker/handlers/accountHandler.py

```diff
@@ -324,6 +325,71 @@ def checkPasswordToken(self,session):
             #failure but alert UI of issue
             return JsonResponse.create(StatusCode.OK,{"errorCode":errorCode,"message":message})
 
+    def updateUser(self, system_email):
+        """
+        Update editable fields for specified user. Editable fields for a user:
+        * is_active
+        * user_status_id
+        * permissions
+
+        Args:
+            None: Request body should contain the following keys:
+                * uid (integer)
+                * status (string)
+                * permissions (comma separated string)
+                * is_active (boolean)
+
+        Returns: JSON response object with either an exception or success message
+
+        """
+        requestDict = RequestDictionary(self.request)
+
+        # throw an exception if nothing is provided in the request
+        if not requestDict.exists("uid") or not (requestDict.exists("status") or requestDict.exists("permissions") or
+                    requestDict.exists("is_active")):
+            # missing required fields, return 400
+            exc = ResponseException("Request body must include uid and at least one of the following: status, permissions, is_active",
+                                    StatusCode.CLIENT_ERROR)
+            return JsonResponse.error(exc, exc.status)
+
+        # Find user that matches specified uid
+        user = self.interfaces.userDb.getUserByUID(int(requestDict.getValue("uid")))
+
+        if requestDict.exists("status"):
+            #check if the user is waiting
+            if(self.interfaces.userDb.checkStatus(user,"awaiting_approval")):
+                if(requestDict.getValue("status") == "approved"):
+                    # Grant agency_user permission to newly approved users
+                    self.interfaces.userDb.grantPermission(user,"agency_user")
+                    link=  AccountHandler.FRONT_END
+                    emailTemplate = { '[URL]':link,'[EMAIL]':system_email}
+                    newEmail = sesEmail(user.email, system_email,templateType="account_approved",parameters=emailTemplate,database=self.interfaces.userDb)
+                    newEmail.send()
+                elif (requestDict.getValue("status") == "denied"):
+                    emailTemplate = {}
+                    newEmail = sesEmail(user.email, system_email,templateType="account_rejected",parameters=emailTemplate,database=self.interfaces.userDb)
+                    newEmail.send()
+            # Change user's status
+            self.interfaces.userDb.changeStatus(user,requestDict.getValue("status"))
+
+        if requestDict.exists("permissions"):
+            permissions_list = requestDict.getValue("permissions").split(',')
+
+            # Remove all existing permissions for user
```

Comment:
On the backend side it's definitely simpler to just grant or remove specified permissions without wiping the rest.  Granting and removing permissions is idempotent, you don't need to care whether they already have the permission.  That said, if the frontend is showing the current permissions and those get maintained unless the user actively unchecks them, I guess this works.